src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/SendOrderContestedAboutToSubmitHandler.java

```diff
@@ -95,41 +108,153 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
             log.info("Share and print general with for Case ID: {}", caseDetails.getId());
             shareAndSendGeneralOrderWithSelectedParties(caseDetails, parties, selectedOrders, printOrderCollection);
 
-
             log.info("Share and print hearing order for Case ID: {}", caseDetails.getId());
-            List<CaseDocument> hearingOrders = generalOrderService.hearingOrdersToShare(caseDetails, selectedOrders);
-            if (hearingOrders != null && !hearingOrders.isEmpty()) {
-                shareAndSendHearingDocuments(caseDetails, hearingOrders, parties, printOrderCollection, userAuthorisation);
+            Pair<List<CaseDocument>, List<CaseDocument>> hearingOrders = generalOrderService.hearingOrdersToShare(caseDetails, selectedOrders);
+            List<CaseDocument> legacyHearingOrders = hearingOrders.getLeft();
+            List<CaseDocument> newProcessedOrders = hearingOrders.getRight();
+
+            if (hasApprovedOrdersToBeSent(legacyHearingOrders, newProcessedOrders)) {
+                // below method also adds the cover sheet even if legacyHearingOrders is empty.
+                shareAndSendHearingDocuments(caseDetails, legacyHearingOrders, parties, printOrderCollection, userAuthorisation);
                 log.info("Sending for stamp final order on Case ID: {}", caseDetails.getId());
-                hearingOrders.forEach(orderToStamp -> {
+                // stamping legacy approved orders and add it to legacy finalised collection
+                legacyHearingOrders.forEach(orderToStamp -> {
                     log.info("StampFinalOrder {} for Case ID: {}, ", orderToStamp, caseId);
                     stampAndAddToCollection(caseDetails, orderToStamp, userAuthorisation);
                 });
+
+                // handling processed orders
+                moveApprovedDocumentsToFinalisedOrder(caseData, newProcessedOrders);
             }
             caseData.setOrdersSentToPartiesCollection(printOrderCollection);
-            caseData.setAdditionalDocument(null);
             setConsolidateView(caseDetails, parties);
-            caseData.setOrdersToShare(new DynamicMultiSelectList());
+
+            clearTemporaryFields(caseData);
+            sendOrdersCategoriser.categorise(caseDetails.getData());
         } catch (RuntimeException e) {
+            log.error(e.getMessage(), e);
```

Comment:
Rather than e.message could you just write out the case id to make it more useful. e.message will appear in the stacktrace.